web/src/search.ts

```diff
@@ -51,12 +51,22 @@ function narrow_or_search_for_term({on_narrow_search}: {on_narrow_search: OnNarr
     }
 
     assert(search_pill_widget !== null);
-    const search_query = search_pill.get_current_search_string_for_widget(search_pill_widget);
+    const search_query = [
+        search_pill.get_current_search_string_for_widget(search_pill_widget),
+        get_search_bar_text(),
+    ]
+        .filter((s) => s !== "")
+        .join(" ");
     if (search_query === "") {
         exit_search({keep_search_narrow_open: true});
         return "";
     }
     const terms = Filter.parse(search_query);
+    // Reset the search bar, in case some of these terms
+    // haven't been pillified yet.
```

Comment:
We need to do this now because `convert_to_pill_on_enter` doesn't. But it's better because here we can do it our custom way.

Reply:
This seems fine, just explain what we're doing and why, with any relevant context on how `convert_to_pill_on_enter` is related to this (Was it already run, but didn't clear the text? Or are we doing this instead of it? Etc.)

Reply:
Yes exactly it's because `convert_to_pill_on_enter`, I'll add a comment